File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int lastSeen[]={-1,-1,-1};
        int l=0,r=0,c=0;
        while(r<s.length()){
            lastSeen[s.charAt(r)-'a']=r;
            if(lastSeen[0]>-1 && lastSeen[1]>-1 && lastSeen[2]>-1){
                c+=Math.min(lastSeen[0],Math.min(lastSeen[1],lastSeen[2]))+1;
            }
            r++;
        }
        return c;
    }
}

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        l=[-1,-1,-1]
        cnt=0
        for i,ch in enumerate(s):
            l[ord(ch)-ord('a')]=i
            if(l[0]!=-1 and l[1]!=-1 and l[2]!=-1):
                cnt+=min(l)+1

        return cnt